Metrics API — Introduce a `/metrics` route for observability / monitoring purposes (#174)
meilisearch/specifications: text/0085-api-keys.md

| tasks.delete     | Provides access to `DELETE` `/tasks` route. |
| settings.get     | Provides access to `GET` `/indexes/:authorizedIndexes/settings` and `/indexes/:authorizedIndexes/settings/*` routes.                                                                                                            |
| settings.update  | Provides access to `POST / DELETE` `/indexes/:authorizedIndexes/settings` and `/indexes/:authorizedIndexes/settings/*` routes.                                                                                                  |
| stats.get        | Provides access to `GET` `/stats`. **⚠️Non-authorized `indexes` are omitted from the response on `/stats`**. Also add access to `GET` `/indexes/:authorizedIndexes/stats`.                                                      |

> | metrics.get      | Provides access to `GET` `/metrics` route. **A restriction on `indexes` does not affect returned metrics.** |

Reviewer: ```suggestion
| metrics.get      | Provides access to `GET` `/metrics` route. **A restriction on `indexes` stops you from calling the route.** |
```

- A restriction on the maximum offset/limit.
- Add search parameters restrictions for an API Key.
- Add rfc2822 format expression for `expiredAt` field. e.g. `Wed, 18 Feb 2022 23:16:09 GMT`

> - Add an alias that can only be associated to one API Key to retrieve it easily on client side. e.g. `GET /keys/:uid_or_alias`

Reviewer 2: So, do we want to add a new action and don't keep the `stats` action?

Author: Since the `stats` endpoint is not removed, we want to keep the `stats` action. However, if we have time to add the `metrics.get` action before the v0.29, I'm all for it, it's cleaner.

Reviewer 2: And do we add it for the admin key rights by default?

Author: Yes!

- Add search parameters restrictions for an API Key.
- Add rfc2822 format expression for `expiredAt` field. e.g. `Wed, 18 Feb 2022 23:16:09 GMT`
- Add an alias that can only be associated to one API Key to retrieve it easily on client side. e.g. `GET /keys/:uid_or_alias`

> - Add a dedicated action for `/metrics` e.g `metrics.get`

Author: ```suggestion
```
